### PR TITLE
Re-enabling file download cache in 2.0

### PR DESCRIPTION
**Pull request by memsharded**
Close https://github.com/conan-io/conan/issues/11658

### CODE REVIEW

File: conans/model/conf.py [hunk 1]
```diff
@@ -49,6 +50,7 @@
     "tools.cmake.cmaketoolchain:toolset_arch": "Toolset architecture to be used as part of CMAKE_GENERATOR_TOOLSET in CMakeToolchain",
     "tools.cmake.cmaketoolchain.presets:max_schema_version": "Generate CMakeUserPreset.json compatible with the supplied schema version",
     "tools.cmake.cmake_layout:build_folder_vars": "Settings and Options that will produce a different build folder and different CMake presets names",
+    "tools.files.download:download_cache": "Define the cache folder to store downloads from files.download()/get()",
```
Comment (czoido) [inline conans/model/conf.py:53]: Why do we need two separate confs? Why not using only one configuration and store all the downloaded files in the same cache?

Comment (memsharded) [reply conans/model/conf.py:53]: There are 2 reasons:
- One technical: the ``conanfile.conf`` received by recipes do not have visibility over the ``core.*`` configuration that is suppossed to be internal only, but not visible or usable by recipes. But having a ``tools.*`` configuration for defining how the internal Conan artifacts downloads are cached seems wrong, as it has nothing to do with the tools, it is something "core"
- The other is functional: there might be users that want to cache only the third-party things downloaded from the internet, but not their own Artifactory files. Or maybe the cleaning policy is different, keeping like forever the files downloaded from ``source()`` from the internet, while cleaning from time to time the cache from internal conan artifacts.

Comment (memsharded) [reply conans/model/conf.py:53]: I am not saying this is immutable, it is true that it is not super-intuitive at first sight. Just trying to explain why it is the way it is. In Conan 1.X is already similar, ``conan.conf`` is the core one, while profiles ``tools.*`` is the downloaded-from-internet one.

Comment (czoido) [reply conans/model/conf.py:53]: I see, I was not aware of that, I guess that users demanded and use that functionality 👍

**Review by czoido — APPROVED** [review]

**Comment by czoido** [comment]
While testing this locally I noticed that if you specify a folder that is not absolute it will put the file download cache inside the folder of the package in the local cache, like this:

image


It's not that is critical, but maybe if it does not make sense to use a folder that is not absolute for this, we should at least warn the user.

**Comment by memsharded** [comment]
> It's not that is critical, but maybe if it does not make sense to use a folder that is not absolute for this, we should at least warn the user.

No, this is not good, I'll fix it

**Comment by memsharded** [comment]
Fixed

**Review by jcar87 — APPROVED** [review]

**Pull request merged**